frontend/src/components/IntegrationConfiguration.tsx

```diff
@@ -176,36 +176,43 @@ const MapStates: FC<{ configuration: IntegrationConfiguration }> = ({
 };
 
 const SelectBoard: FC<{ configuration: IntegrationConfiguration }> = ({
-  configuration: { id, name, roadmapId, boardId },
+  configuration,
 }) => {
+  const { id, name, roadmapId, boardId } = configuration;
   const { t } = useTranslation();
 
   const [errorMessage, setErrorMessage] = useState('');
   const [patch] = apiV2.usePatchIntegrationConfigurationMutation();
 
-  const [tokenError, setTokenError] = useState(false);
+  const [boardsTokenError, setBoardsTokenError] = useState(false);
+  const [selectedError, setSelectedError] = useState(false);
   const selected = apiV2.useGetIntegrationSelectedBoardQuery(
     { name, roadmapId },
     { skip: !boardId },
   );
 
   const boards = apiV2.useGetIntegrationBoardsQuery(
     { name, roadmapId },
-    { skip: selected.isError },
+    { skip: !selected.isSuccess },
```

Comment:
```suggestion
    { skip: !!boardId && !selected.isSuccess },
```

Noticed that this fails if when there is no board selected yet.

If there is no board selected, the selected board query is skipped, but this should be tried.

The queries should still remain sequential, since the selected query was skipped